tools/cli/adminDBCleanCommand.go

```diff
@@ -205,26 +208,23 @@ func cleanShard(
 			WorkflowID: ce.WorkflowID,
 			RunID:      ce.RunID,
 		}
-		preconditionForDBCall(&report.TotalDBRequests, limiter)
-		err = execStore.DeleteWorkflowExecution(deleteConcreteReq)
+		err = retryDeleteWorkflowExecution(limiter, &report.TotalDBRequests, execStore, deleteConcreteReq)
 		if err != nil {
 			report.Handled.FailedCleanedCount++
 			failedCleanWriter.Add(&ce)
 			continue
 		}
 		report.Handled.SuccessfullyCleanedCount++
 		successfullyCleanWriter.Add(&ce)
-		if ce.CorruptedExceptionMetadata.CorruptionType != OpenExecutionInvalidCurrentExecution {
```

Comment:
Deleted because you want to rely on the fact that `DeleteCurrentWorkflowExecution` is a conditional delete based on the runID? Just want to confirm.

Reply:
Yes